lms/views/admin/application_instance.py

```diff
@@ -9,17 +9,63 @@
 from lms.security import Permissions
 from lms.services import ApplicationInstanceNotFound, LTIRegistrationService
 from lms.validation._base import PyramidRequestSchema
-from lms.views.admin import flash_validation
+from lms.views.admin import error_render_to_response, flash_validation
+
+
+class EmptyStringNoneMixin:
+    """
+    Allows empty string as "missing value".
+
+    Marshmallow doesn't have a clean solution yet to POSTed values
+    (that are always present  in the request as empty strings)
+
+    Here we convert them explicitly to None
+
+    https://github.com/marshmallow-code/marshmallow/issues/713
```

Comment:
I thought it was missing something but this seems like big omission, can't really have optional fields in a form without this.

In any case a custom field it's not the end of the world.